cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/RemoteDatasetFramework.java

```diff
@@ -199,6 +203,7 @@ private void addModule(Id.DatasetModule moduleId, Class<?> typeClass) throws Dat
     try {
       File tempFile = File.createTempFile(typeClass.getName(), ".jar");
       try {
+        // TODO: should be namespaced?
```

Comment:
Its created during the deployment of a module within a namespace. And we have a ``/cdap/[namespace]/tmp`` directory for such temporary files, etc. I had missed the TODO. Will address.